src/Neo/Network/UPnP.cs

```diff
@@ -183,8 +183,8 @@ private static XmlDocument SOAPRequest(string url, string soap, string function)
             request.Headers.Add("Content-Type", "text/xml; charset=\"utf-8\"");
             request.Content = new StringContent(req);
             using HttpClient http = new();
-            using HttpResponseMessage response = http.Send(request);
-            using Stream stream = response.EnsureSuccessStatusCode().Content.ReadAsStream();
+            using HttpResponseMessage response = http.SendAsync(request).GetAwaiter().GetResult();
+            using Stream stream = response.EnsureSuccessStatusCode().Content.ReadAsStreamAsync().GetAwaiter().GetResult();
```

Comment:
Please make async.
```csharp
using HttpResponseMessage response = await http.SendAsync(request);
using Stream stream = await response.EnsureSuccessStatusCode().Content.ReadAsStreamAsync();
```

Reply:
Why we should change it?

Reply:
instead of changing everywhere. You add sync in the P2P protocol function. But doesn't really matter, just looks better.

Reply:
The problem is that then we need to set the method as async, and when we call it, do the same. I think that we have async code that it should be sync, like the smart contract execution